postreise/analyze/generation/tests/test_summarize.py

```diff
@@ -53,10 +54,10 @@
 
 grid_attrs = {"plant": mock_plant, "bus": mock_bus, "storage_gen": mock_storage}
 scenario = MockScenario(grid_attrs, pg=mock_pg, storage_pg=mock_storage_pg)
-scenario.state.grid.zone2id = {
-    "Washington": 201,
-    "Oregon": 202,
+scenario.state.grid.id2zone = {
+    k: v for k, v in zip(mock_plant["zone_id"], mock_plant["zone_name"])
 }
+scenario.state.grid.zone2id = {v: k for k, v in scenario.state.grid.id2zone.items()}
```

Comment:
I was going to suggest we build these mappings into the `MockGrid` class, but I think we kinda have them already? 
```
In [15]: mg = MockGrid()

In [16]: mg.model_immutables.zones.keys()
Out[16]: dict_keys(['abv', 'abv2id', 'abv2interconnect', 'abv2loadzone', 'abv2state', 'id2abv', 'id2loadzone', 'id2timezone', 'interconnect', 'interconnect2abv', 'interconnect2id', 'interconnect2loadzone', 'interconnect2timezone', 'interconnect_combinations', 'loadzone', 'loadzone2id', 'loadzone2interconnect', 'loadzone2state', 'mappings', 'state', 'state2abv', 'state2loadzone', 'timezone2id'])

```

So if we want to use these by default we could set the `id2zone` and `zone2id` attributes in `MockGrid`, or could use the mappings from `mg.model_immutables` to set the values here.

Reply:
The scenario here is a MockScenario which uses a MockGrid already. But we are not using `model_immutables` though given those mappings are set in the grid object to mimic the behavior of a real grid object.

Reply:
A couple of comments:

First, this is not currently well designed. If you do:
```
>>> from powersimdata import Grid
>>> g = Grid('Texas')
Reading bus.csv
Reading plant.csv
Reading gencost.csv
Reading branch.csv
Reading dcline.csv
Reading sub.csv
Reading bus2sub.csv
Reading zone.csv
>>> g.zone2id
{'Far West': 301, 'North': 302, 'West': 303, 'South': 304, 'North Central': 305, 'South Central': 306, 'Coast': 307, 'East': 308}
```
So the `zone2id` attribute of a `Grid` object relates the load zone name to the load zone id in the interconnect used for the instantiation.

If you do:
```
g.model_immutables.zones["interconnect2abv"]
```
you get all the mappings for the grid model (USA), not only for the interconnect (here ERCOT)

I believe it would make sense to trim the nappings accessed through the `model_immutables` attribute of the `Grid` object to reflect the interconnect used for the instantiation.

Then, I would not used the mappings in `mg.model_immutables` since those will have the real mappings of the model while one is free to define `zone_name` as `[A, B, C]` and `zone_id` as `[1,2,3]` in the mock plant. By the way `zone_id` was defined as `[1, 2, 3]` before I refactored it and this is in contradiction to what is in `mg.model_immutables`

Reply:
@jon-hagg, there is https://github.com/Breakthrough-Energy/PowerSimData/issues/492 that would help the mocking process. Do you want to tackle it?